src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/common/stateful_init_graph.cpp

```diff
@@ -0,0 +1,199 @@
+                                   public CPUTestsBase {
+public:
+    static std::string getTestCaseName(const testing::TestParamInfo<InitGraphStatefulModelTestParams>& obj) {
+        std::ostringstream result;
+        std::vector<InputShape> inputShapes = obj.param;
+
+        result << "IS=";
+        for (const auto& shape : inputShapes) {
+            result << ov::test::utils::partialShape2str({shape.first}) << "_";
+        }
+        result << "TS=";
+        for (const auto& shape : inputShapes) {
+            result << "(";
+            if (!shape.second.empty()) {
+                for (const auto& itr : shape.second) {
+                    result << ov::test::utils::vec2str(itr);
+                }
+            }
+            result << ")";
+        }
+        result << ")";
+
+        return result.str();
+    }
+
+    std::vector<ov::Tensor> calculate_refs() override {
+        for (const auto& param : functionRefs->get_parameters()) {
+            inferRequestRef.set_tensor(param->get_default_output(), inputs.at(matched_parameters[param]));
+        }
+        inferRequestRef.infer();
+
+        auto outputs = std::vector<ov::Tensor>{};
+        for (const auto& output : functionRefs->outputs()) {
+            outputs.push_back(inferRequestRef.get_tensor(output));
+        }
+
+        return outputs;
+    }
+
+    std::vector<ov::Tensor> get_plugin_outputs() override {
+        for (const auto& input : inputs) {
+            inferRequest.set_tensor(input.first, input.second);
+        }
+        inferRequest.infer();
+        auto outputs = std::vector<ov::Tensor>{};
+        for (const auto& output : function->outputs()) {
+            outputs.push_back(inferRequest.get_tensor(output));
+        }
+        return outputs;
+    }
+
+    void run() override {
+        prepare();
+
+        for (const auto& targetStaticShapeVec : targetStaticShapes) {
+            for (auto iters = 0; iters < 2; iters++) {
+                generate_inputs(targetStaticShapeVec);
+                validate();
+            }
+            // Different input shape, reset is required.
+            reset();
+        }
+    }
+
+protected:
+    void reset() {
+        for (auto&& state : inferRequest.query_state()) {
+            state.reset();
+        }
+
+        for (auto&& state : inferRequestRef.query_state()) {
+            state.reset();
+        }
+    }
+
+    void prepare() {
+        compile_model();
+
+        inferRequest = compiledModel.create_infer_request();
+        ASSERT_TRUE(inferRequest);
+
+        // Node with friendly name "init_graph/add_1" should be moved into subgraph.
+        bool found_init_graph_add = false;
+        for (auto node : compiledModel.get_runtime_model()->get_ops()) {
+            if (node->get_friendly_name() == "init_graph/add_1") {
+                found_init_graph_add = true;
+                break;
+            }
```

Comment:
Should we add the same check for the matmul?

Reply:
Yes, added.